### ISSUE TITLE
upload progress reports wrong numbers

### ISSUE THREAD

**Issue by VilemKurz** [priore/SOAPEngine#96]
Upload progress in either delegate callback
```swift
optional public func soapEngine(soapEngine: SOAPEngine!, didSendDataSize current: UInt, total: UInt)
```
 or in block method
```swift 
public func requestURL(asmxURL: AnyObject!, soapAction: String!, value: AnyObject!, forKey key: String!, completeWithDictionary complete: SOAPEngineCompleteBlockWithDictionary!, failWithError fail: SOAPEngineFailBlock!, receivedDataSize receive: SOAPEngineReceiveDataSizeBlock!, sendedDataSize sended: SOAPEngineSendDataSizeBlock!)
```
reports wrong numbers in *total* parameter. 
**Actual result**: returns increasing number each time - in fact it is total uploaded bytes so far. This number I would expect to be returned from *current* parameter
**Expected result**: return total number of bytes to be uploaded (e.g. whole file size) as *total* and already uploaded bytes as *current*

Note: download blocks work as expected. I suggest to unify this behaviour.

**Comment by priore** [comment]
This is really strange, the code we use for the generation of the delegates and of the block is very simple, for received data is:

```objective-c
// contentLength return the Content-Length value received on header of response (taken one).
// webData is the incremental packet of data received (NSData).
receiveBlock([webData length], contentLength);
[self.delegate soapEngine:self didReceiveDataSize:[webData length] total:contentLength];
```
instead for sending data is:
```objective-c
// bytesWritten and totalBytesWritten have same values of the didSendBodyData delegate of NSURLConnection.
sendBlock(bytesWritten, totalBytesWritten);
[self.delegate soapEngine:self didSendDataSize:bytesWritten total:totalBytesWritten];
```
you have a service where we can test the problem you're having?
thanks

**Comment by priore** [comment]
we found the problem, we must use _totalBytesExpectedToWrite_ and not _totalBytesWritten_, thanks for your test and for your support.

**Comment by priore** [comment]
We just released the new version 1.26 of the framework 017ccf18a1667c28611cb529fb9e6efb040fa516
